### PR TITLE
Fix digest credentials for proxy in windows

### PR DESCRIPTION
**Pull request by cbargren**
Using a proxy with digest credentials was not working in Windows. It was simply always using basic auth whenever a plaintext credential mechanism was specified. Detecting whether the plaintext credential mechanism was digest or basic and applying that mechanism to `WinHttpSetCredentials` fixed the problem.

### CODE REVIEW

File: src/transports/winhttp.c [hunk 1]
```diff
@@ -1072,7 +1083,7 @@ static int winhttp_stream_read(
 			/* TODO: extract the username from the url, no payload? */
 			if (t->owner->proxy.credentials) {
 				int cred_error = 1;
-				cred_error = t->owner->proxy.credentials(&t->proxy_cred, t->owner->proxy.url, NULL, allowed_types, NULL);
+				cred_error = t->owner->proxy.credentials(&t->proxy_cred, t->owner->proxy.url, NULL, allowed_types, t->owner->proxy.payload);
```
Comment (ethomson) [inline src/transports/winhttp.c]: I don't quite understand where this change is coming from yet.  I don't immediately see anything that's using a payload with regards to the digest changes...  Is this a related change, or should it be split out into its own commit?

Comment (cbargren) [inline src/transports/winhttp.c]: Without the payload, the `credentials` call was segfaulting in https://github.com/nodegit/nodegit. It isn't directly related to digest credentials, but without it, the credentials callback isn't working. I'll go ahead and split it into its own commit.

Comment (ethomson) [inline src/transports/winhttp.c]: Got it.  Thanks for clarifying; I do appreciate that you split this out into its own commit.
